sktime/transformations/series/rounding.py

```diff
@@ -0,0 +1,303 @@
+        self.round_to_dp = round_to_dp
+        self.round_to_multiple = round_to_multiple
+        self.round_to_list = round_to_list
+
+        self._default: bool = not any((round_to_dp, round_to_multiple, round_to_list))
+
+        # Base class init
+        super(Discretizer, self).__init__()
+
+        # validate arguments
+        self._check_arguments()
+
+    def _check_method(self):
+        """
+        Check that the inputs are valid for the transformation.
+
+        Warns
+        -----
+        UserWarning
+            (1) If more than one of round_to_dp, round_to_multiple, or round_to_list are not None.
+
+        Raises
+        ------
+        ValueError
+            (1)  `round_to_dp` is not None and is not an int or float.
+            (2)  `round_to_multiple` is not None and is not an int or float.
+            (3)  `round_to_list` is not None and is not an int or float.
+            (4)  `round_to_list` is not None and any element is not an int or float.
+        """
+        if sum((bool(i) for i in (
+                self.round_to_dp,
+                self.round_to_list,
+                self.round_to_multiple
+                ))) > 1:
+            # Raise warning if more than one argument passed
+            raise UserWarning(
+                """
+                    More than one argument has been passed,
+                    only one argument can be applied.
+                    These are evaluated/applied in the order:
+                    `round_to_dp` , `round_to_list`, `round_to_multiple`.
+                    """)
+
+        if self.round_to_dp is not None and not isinstance(
+                self.round_to_dp, (int, float)):
+            raise ValueError(
+                f"""
+                round_to_dp: {self.round_to_dp} is not an integer or float.
+                """)
+
+        if self.round_to_multiple is not None and not isinstance(
+                self.round_to_multiple, (int, float)):
+            raise ValueError(
+                f"""
+                round_to_multiple: {self.round_to_multiple} is not an integer or float.
+                """
+            )
+        if self.round_to_list is not None and not isinstance(
+                self.round_to_list, list):
+            raise ValueError(
+                f"""
+                round_to_list: {self.round_to_list}, {type(self.round_to_list)} is not a list.
+                """)
+        if self.round_to_list is not None and not all(
+            (
+                isinstance(i, (int, float)) for i in self.round_to_list)):
+            wrong_types: list = [
+                type(i) for i in self.round_to_list if not isinstance(
+                    i, (int, float))]
+            raise ValueError(
+                f"""
+                round_to_list: {self.round_to_list}, not all elements are integer or float.
+                {wrong_types} were found in the list.
+                """)
+
+    def _fit(self, X, y=None):
+        return super()._fit()
```

Comment:
unnecessary, and will likely break, since `_fit` has no args. Just use inheritance.

Reply:
I had actually just left these in as the linter was complaining about abstract classes... sorry about that.